src/Workspaces/Core/Portable/Utilities/TaskExtensions.cs

```diff
@@ -50,7 +54,7 @@ private static bool IsThreadPoolThread(Thread thread)
 
         public static T WaitAndGetResult<T>(this Task<T> task, CancellationToken cancellationToken)
         {
-#if DEBUG
+#if DEBUG && WORKSPACE
```

Comment:
Ugh this really worries me. Mostly because if we lost this #define somehow we've just lost some pretty critical protection and we won't ever know it.

Reply:
Hmm, turns out it's not needed anymore. Trying to remove the extra condition. This is a big reflection hack though :(

Reply:
Oh, I see what changed. @sharwell switched from 

```
var threadKind = ForegroundThreadDataInfo.CurrentForegroundThreadDataKind;
if (threadKind == ForegroundThreadDataKind.Unknown)
```

to the reflection hack. It works now outside of Workspace. On the other hand, it's a reflection hack :(